nucypher/cli/commands/dkg.py

```diff
@@ -0,0 +1,343 @@
+import random
+import time
+
+import click
+import maya
+from nucypher_core.ferveo import DkgPublicKey
+from web3 import Web3
+
+from nucypher.blockchain.eth.agents import (
+    ContractAgency,
+    CoordinatorAgent,
+    PREApplicationAgent,
+)
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+from nucypher.blockchain.eth.signers import Signer
+from nucypher.characters.lawful import Bob, Enrico
+from nucypher.cli.options import option_eth_provider_uri
+from nucypher.crypto.powers import TransactingPower
+from nucypher.policy.conditions.lingo import ConditionLingo
+from nucypher.utilities.emitters import StdoutEmitter
+from nucypher.utilities.logging import GlobalLoggerSettings
+
+GlobalLoggerSettings.start_console_logging()
+emitter = StdoutEmitter(verbosity=2)
+
+
+option_eth_staking_network = click.option(
+    "--eth-staking-network",
+    "eth_staking_network",
+    help="ETH staking network",
+    type=click.Choice(["tapir", "lynx"]),
+    default="lynx",
+)
+
+option_coordinator_provider = click.option(
+    "--coordinator-provider",
+    "coordinator_provider_uri",
+    help="Coordinator network provider URI",
+    type=click.STRING,
+    required=True,
+)
+
+option_coordinator_network = click.option(
+    "--coordinator-network",
+    "coordinator_network",
+    help="Coordinator network",
+    type=click.Choice(["mumbai"]),
+    default="mumbai",
+)
+
+
+def get_agency(
+    coordinator_network, coordinator_provider_uri, eth_staking_network, eth_provider_uri
+):
+    coordinator_agent = ContractAgency.get_agent(
+        agent_class=CoordinatorAgent,
+        registry=InMemoryContractRegistry.from_latest_publication(
+            network=coordinator_network
+        ),
+        provider_uri=coordinator_provider_uri,
+    )  # type: CoordinatorAgent
+
+    staking_network_registry = InMemoryContractRegistry.from_latest_publication(
+        network=eth_staking_network
+    )
+    application_agent = ContractAgency.get_agent(
+        agent_class=PREApplicationAgent,
+        registry=staking_network_registry,
+        provider_uri=eth_provider_uri,
+    )  # type: PREApplicationAgent
+
+    return coordinator_agent, application_agent, staking_network_registry
+
+
+@click.group()
+def dkg():
```

Comment:
I think in general we should about exposing the "DKG" term, which is very obscure. What do you think about `rituals`?

Reply:
✔️ - went with `ritual`.